include/timer.h

```diff
@@ -0,0 +1,23 @@
+#ifndef _TIMER_H
+#define _TIMER_H
+#include "common.h"
+
+#define MAX_TIMER   128
+
+#define UNUSE   0
+#define USED    1
+
+typedef struct timer{
+    struct timer *next;
+    unsigned int timeout, flags;
+    int data;
+}timer;
+
```

Comment:
to be simple, i think following struct already can reach our goal. 
timer {
    struct timer *next;
    long timeout;
}

Reply:
your opinions look like a work queue.